### PR TITLE
complete feta-22

### PR DESCRIPTION
**Pull request by imyungchu**
In the report, we have learned:
1.  In #10 , we know how to connect our GitHub account to a remote server with public key and private key, also know about the famous ED25519 and RSA encryption algorithm.
2. In #12 , we install a powerful `Nginx` on Linux server, and deployed our previous app on `nycu.me`. We dive in to study more about the command such as `npm run dev -- --host`, `run dev`, and `run build`. We write some ideas about differences between port number, development environment, and production environment.
3. In #11 , we got a whole picture and clear flow about how we can finish a task in a template and some `git` commands. We also answer some inspiring questions in [`report-feta-22.md`](imyungchu/report-fetp-22.md)

### CODE REVIEW

File: report-templates/report-fetp-22.md
Comment (kdotwei) [inline report-templates/report-fetp-22.md:1]: Template 不能改到！再重新將 main merge 回你的 branch，並讓 template 保持 main 版本，其他的 conflict 保持你的版本！有問題這邊留言討論！

Comment (imyungchu) [reply report-templates/report-fetp-22.md:1]: 感謝提醒！不知道要怎麼從 main merge 回 branch 所以選擇笨笨的還原檔案

File: imyungchu/report-fetp-22.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# FETP-22 Report
+
+## INFORMATION
+**GitHub ID**: `imyungchu`
+
+## FETP-21
+
+> **QUESTION:** 請簡單說明非對稱加密的概念。並說明 Public Key 與 Private Key 的用法。
+
+非對稱加密是利用那個人公開的 public key 可以加密，但是只有那個人 Private Key 才能解開的問題，進而實現加密發送訊息給同學的功能。
```
Comment (kdotwei) [inline imyungchu/report-fetp-22.md:10]: 反過來就是數位簽章！

File: imyungchu/report-fetp-22.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+1. 如果 key 選擇質數的方式不夠隨機，就可能會被因數分解，破解加密。可以用偽隨機產生器來避免。
+2. 如果 key 太小或彼此太相近，也很可能會被猜出，所以安全專家會建議最小 key 要大於 2048 bits。
+
+---
+
+## FETP-23
+
+> **QUESTION:** 為何我們安裝了 Nginx 後，在瀏覽器輸入網址或 IP 就可以連到網站？而在先前的任務中都要加入 Port 號？
+
+Nginx 可以讓我們在同一個 port 的主機，在不用輸入 port 的情況下運行，因為他是一個可以自動幫我們回應網站 request 的 web server。支援不同網址共享 80 port 實現一個主機同時跑多個網站，也就是一個「反向代理」的伺服器。
```
Comment (kdotwei) [inline imyungchu/report-fetp-22.md:36]: 嚴格說起來是 Nginx 一裝好，預設會監聽 80 Port，而 80 Port 正是 http 協議的預設 Port。當你使用 http 連線至指定的伺服器時，瀏覽器會幫你向伺服器的 80 Port 發送 Request。

**Review by kdotwei — COMMENTED** [review]
一些留言，可以把他們加入到 Report 中！

File: imyungchu/report-fetp-22.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+
+反向代理的觀念類似打電話去客服，我們會得到服務但不見得是同一個客服回答，但客服公司會去分派任務讓每個 request 都盡量有回應。之前要輸入 port 時，可能會有網址重疊問題，現在有了 Nginx 我們可以統一交由 Nginx 處理配置問題，進而幫助網站伺服器更有效率的連接。
```
Comment (iriszzzz) [inline imyungchu/report-fetp-22.md:39]: 真的好喜歡你的比喻 !!

**Review by iriszzzz — COMMENTED** [review]
詠筑的 report 每次都好讚 ❤️ !

File: imyungchu/report-fetp-22.md (hunk 3 above)
Comment (kdotwei) [reply imyungchu/report-fetp-22.md:39]: @iriszzzz 喜歡然後不給人家 approve 好狠 na

Comment (iriszzzz) [reply imyungchu/report-fetp-22.md:39]: 怎麼有人在挑撥阿==

**Review by iriszzzz — APPROVED** [review]
🙌 🙌

**Pull request merged**